src/cryptography/x509/extensions.py

```diff
@@ -342,6 +342,9 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self == other
 
+    def __hash__(self):
+        return hash(self.ca) ^ hash(self.path_length)
```

Comment:
So there's this (xoring each of the hashes), or `hash((self.ca, self.path_length))` (xoring a tuple of the items).

https://github.com/python/cpython/blob/master/Objects/tupleobject.c#L337-L359 is the algorithm for tuples, basically a xor + fancy stuff.

opinions?